### PR TITLE
chore: replace grpc-web with @improbable-eng/grpc-web

### PR DESCRIPTION
**Pull request by markin-io**
### TLDR
Chrome Web Store will not allow publishing Extensions that are using [grpc-web](https://github.com/grpc/grpc-web) because under the hood it operates with `XMLHttpRequest` and it's not supported by Manifest V3. [@improbable-eng/grpc-web](https://github.com/improbable-eng/grpc-web/) uses `Fetch API` under the hood which is supported by Manifest V3

### Detailed explanation.

In July Google [stopped allowing](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/) publishing of Extensions operating on Manifest V2.

Migration from V2 to V3 requires usage of the [service worker](https://developer.chrome.com/docs/extensions/mv3/mv3-migration/#man-sw) in the background instead of a simple background script. 

Service workers in Web Extensions background [do not support](https://developer.chrome.com/docs/extensions/mv3/mv3-migration/#background-service-workers) `XMLHttpRequest` and can only operate with the `Fetch API`.

Finally, [grpc-web](https://github.com/grpc/grpc-web) operates with the `XMLHttpReqiest` and [@improbable-eng/grpc-web](https://github.com/improbable-eng/grpc-web/) operates with `Fetch API`

## What was done?

- Switched `grpc-web` to `@improbable-eng/grpc-web`
- Re-generated protocol buffer Web JS clients 
- Added compatibility layer to match with the `grpc/grpc-js`
- Updated tests

## How Has This Been Tested?

- CI (specifically, karma tests)
- Ran in Web Extension that operates with Manifest V3

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by shumkov** [comment]
We probably don't want to introduce such radical change to v0.23 which more or less stable and almost ready to release

**Comment by strophy** [comment]
Hi @markin-io can you please fill in the description of this PR so I can understand the purpose of this change?

File: packages/dapi-grpc/browser.js [hunk 1]
```diff
@@ -1,10 +1,32 @@
...
+const coreMessages = require('./clients/core/v0/web/core_pb');
+const platformMessages = require('./clients/platform/v0/web/platform_pb');
+
+const parseMetadata = (metadata) => {
+  if (metadata instanceof Metadata) {
+    const parsedMetadata = {};
+    metadata.forEach((key, values) => {
+      // Join with comma because metadata items
+      // represented by an array of browser headers
+      parsedMetadata[key] = values.join(', ');
```
Comment (markin-io) [inline packages/dapi-grpc/browser.js:14]: Make sure it's compliant with grpc-js Metadata

File: packages/js-dapi-client/lib/transport/GrpcTransport/createGrpcTransportError.js [hunk 2]
```diff
@@ -56,7 +56,14 @@ function createGrpcTransportError(grpcError, dapiAddress) {
     // In cases of gRPC-Web client we get plain map instead of Metadata instance
     metadata = grpcError.metadata;
     if (grpcError.metadata instanceof Metadata) {
+      // Handle grpc-js metadata
       metadata = grpcError.metadata.getMap();
+    } else if (grpcError.metadata.headersMap) {
```
[the remark this answers is not in the document]
Comment (markin-io) [reply packages/js-dapi-client/lib/transport/GrpcTransport/createGrpcTransportError.js]: Fixed

File: packages/js-dash-sdk/src/SDK/Client/Platform/broadcastStateTransition.ts [hunk 3]
```diff
@@ -74,7 +74,8 @@ export default async function broadcastStateTransition(
 
     if (error) {
         // Create DAPI response error from gRPC error passed as gRPC response
-        const grpcError = new GrpcError(error.code, error.message, error.data);
+        const grpcError = new GrpcError(error.code, error.message);
+        grpcError.metadata = error.data;
```
Comment (shumkov) [inline packages/js-dash-sdk/src/SDK/Client/Platform/broadcastStateTransition.ts:78]: Could you please add a comment / todo to explain this

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]
Looks good as a first iteration 👍

**Review by shumkov — APPROVED** [review]
LGTM 👍
